app.rb

```diff
@@ -1,29 +1,191 @@
+    validate_option
+  end
+
+  def validate_option
+    option = gets.chomp.to_i
+
+    menu_actions = {
+      1 => :list_all_books,
+      2 => :list_all_people,
+      3 => :create_person,
+      4 => :create_book,
+      5 => :create_rental,
+      6 => :list_rentals_by_person_id,
+      7 => :exit
+    }
+
+    action = menu_actions[option]
+
+    if action
+      send(action)
+    else
+      puts 'Select a valid option'
+      display_menu
+    end
+  end
+
+  # Person methods
+  def list_all_people
+    check_empty_list(@people, 'people')
+
+    @people.each do |person|
+      puts "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+
+    display_menu
+  end
+
+  def create_student
+    print 'Age: '
+    age = gets.chomp
+
+    print 'Name: '
+    name = gets.chomp
+
+    @people << Student.new(age, nil, name)
+    puts 'New person (Student) created successfully'
+
+    display_menu
+  end
+
+  def create_teacher
+    print 'Age: '
+    age = gets.chomp
+
+    print 'Name: '
+    name = gets.chomp
+
+    print 'Specialization: '
+    specialization = gets.chomp
+
+    @people << Teacher.new(age, specialization, name)
+    puts 'New person (Teacher) created successfully'
+
+    display_menu
+  end
+
+  def create_person
+    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+    input = gets.chomp
+    case input
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    else
+      puts 'Please enter a valid number (1 or 2)'
+
+      create_person
+    end
+  end
+
+  # Book methods
+  def list_all_books
+    check_empty_list(@books, 'books')
+
+    @books.each do |book|
+      puts "Title: #{book.title}, Author: #{book.author}"
+    end
+
+    display_menu
+  end
+
+  def create_book
+    print 'Title: '
+    title = gets.chomp
+
+    print 'Author: '
+    author = gets.chomp
+
+    @books << Book.new(title, author)
+    puts 'Book created successfully'
+
+    display_menu
+  end
+
+  # Rental methods
+  def create_rental
+    puts 'Select a book from the following list by number: '
+
+    check_empty_list(@books, 'books')
+    @books.each_with_index do |book, index|
+      puts "#{index}) Title: #{book.title}, Author: #{book.author}"
+    end
+    book_index = gets.chomp.to_i
+
+    puts 'Select a person from the following list by number (Not ID): '
+
+    check_empty_list(@people, 'people')
+    @people.each_with_index do |person, index|
+      puts "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+    person_index = gets.chomp.to_i
+
+    print 'Date: '
+    date = gets.chomp
+
+    @rentals << Rental.new(date, @people[person_index], @books[book_index])
+    puts 'Rental created successfully'
+
+    display_menu
+  end
+
+  def list_rentals_by_person_id
+    print 'ID of the person: '
+    id = gets.chomp.to_i
+
+    rentals = @rentals.select do |rental|
+      rental.person.id.to_i == id
+    end
```

Comment:
- [ Required ] I think it would be nice if you fix the error that occurs when I try to list the rentals that a given user has, currently When I try to list the rentals that a given user has using their `id` I get the error below. Please see the screenshot below for more details on this and rectify it.

## Error when running project
![Screenshot from 2023-09-02 02-58-09](https://github.com/mwismann/oop_school_library/assets/16392046/935619f7-1b83-4161-84a9-4b50fce15dfe)